### ISSUE TITLE
crash with PythonQt::init

### ISSUE THREAD

**Issue by wangdaye078** [MeVisLab/pythonqt#212]
Qt5.15.13 + python3.8 + pythonqt3.4.2 is ok,3.5.0+ is crash

```
PythonQt-Qt5-Python3.8_d.dll!PythonQtConv::convertQtValueToPythonInternal(int type, const void * data) 行 224	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtConv::QVariantToPyObject(const QVariant & v) 行 1399	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQt::addVariable(_object * object, const QString & name, const QVariant & v) 行 1171	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtObjectPtr::addVariable(const QString & name, const QVariant & v) 行 77	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtClassInfo::createEnumWrappers(const QMetaObject * meta) 行 884	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtClassInfo::createEnumWrappers(const QObject * decoratorProvider) 行 900	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtClassInfo::decorator() 行 757	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtClassInfo::findEnumWrapper(const char * name) 行 915	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtClassInfo::findEnumWrapper(const QByteArray & name, PythonQtClassInfo * localScope, bool * isLocalEnum) 行 852	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtMethodInfo::fillParameterInfo(PythonQtMethodInfo::ParameterInfo & type, const QByteArray & orgName, PythonQtClassInfo * classInfo) 行 220	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtMethodInfo::PythonQtMethodInfo(const QMetaMethod & meta, PythonQtClassInfo * classInfo) 行 67	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtMethodInfo::getCachedMethodInfo(const QMetaMethod & signal, PythonQtClassInfo * classInfo) 行 105	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQtPrivate::addDecorators(QObject * o, int decoTypes) 行 1612	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQt::addDecorators(QObject * o) 行 1495	C++
PythonQt-Qt5-Python3.8_d.dll!PythonQt::init(int flags, const QByteArray & pythonQtModuleName) 行 226	C++
```

**Comment by mrbean-bremen** [comment]
@wangdaye078  - so what did you do, and what was the actual crash error message? 

And incidentally, I guess 行 means something like "line number"?

**Comment by mrbean-bremen** [comment]
@usiems - this crashes while adding enum values to the type itself (which has been newly introduced), it looks like some kind of conversion is not handled in `convertQtValueToPythonInternal`, though I'm not sure. Can you have a look? This looks like a possible regression.

**Comment by wangdaye078** [comment]
sorry i am chinese, my english is poor, write english with google translate.
I found out that this problem was caused by my modification.
Because I couldn't require the python version used by the user, I modified the PythonQt code so that I could first find the python version used by the user during the execution process, and then load the corresponding pythonQt library. I wrote a script to modify the code, The code is at https://github.com/wangdaye078/pythonQtCustomModification/blob/main/pythonQtCustomModification.py.
In PythonQtObjectPtr.h, I changed the "PyObject* object() const" function of PythonQtObjectPtr to "virtual PyObject* object() const", which caused the crash.
